tests/e2e/e2e.sh

```diff
@@ -17,83 +17,57 @@
 set -eux
 
 export ARTIFACTS="${ARTIFACTS:-$(mktemp -d)}"
+ROOT=$(cd ../../../; pwd)
 
-function cleanup_kind_cluster() {
-    kind export logs --name istio-testing "${ARTIFACTS}/kind"
-    kind delete cluster --name=istio-testing
-}
-
-function setup_kind_cluster() {
-  # Delete any previous e2e KinD cluster
-  echo "Deleting previous KinD cluster with name=istio-testing"
-  if ! (kind delete cluster --name=istio-testing) > /dev/null; then
-    echo "No existing kind cluster with name istio-testing. Continue..."
-  fi
-
-  trap cleanup_kind_cluster EXIT
-
-  # Create KinD cluster
-  if ! (kind create cluster --name=istio-testing --loglevel debug --retain --wait 30s); then
-    echo "Could not setup KinD environment. Something wrong with KinD setup. Exporting logs."
-    exit 1
-  fi
-
-  KUBECONFIG="$(kind get kubeconfig-path --name="istio-testing")"
-  export KUBECONFIG
-
-cat <<EOF > metallb-config.yaml
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  namespace: metallb-system
-  name: config
-data:
-  config: |
-    address-pools:
-    - name: default
-      protocol: layer2
-      addresses:
-      - 172.17.255.1-172.17.255.250
-EOF
-
-  kubectl apply -f https://raw.githubusercontent.com/google/metallb/v0.8.1/manifests/metallb.yaml
-  kubectl apply -f ./metallb-config.yaml
-}
 
 function setup_docker() {
-  HUB=istio-testing TAG=latest make -f Makefile.core.mk controller docker
-  kind --loglevel debug --name istio-testing load docker-image istio-testing/operator:istio-testing
+  HUB=istio-testing TAG=1.5-dev make controller docker
+  kind --loglevel debug --name istio-testing load docker-image istio-testing/operator:1.5-dev
+  kind --loglevel debug --name istio-testing load docker-image istio-testing/app:istio-testing
 }
 
-setup_kind_cluster
-setup_docker
+
 
 mkdir -p "${ARTIFACTS}/out"
 
-ISTIO_DIR="${GOPATH}/src/istio.io/istio"
+ISTIO_DIR="${ROOT}/src/istio.io/istio"
```

Comment:
this is definately wrong - although it may be my original work where it was wrong.

I'd recommend trying `ISTIO_DIR=$(mktemp -d)`.  If that fails because of problems in Istio, we will need to work through that problem (we can do that after this PR merges) because hardcode paths in the build system are undesireable.

Reply:
Further following this up - the git clone below could negatively interact with a local copy of istio (I recognize a test for the directory is done) but this would result in $ISTIO/istio being used from the local filesystem if the developer has work in that location.  This is highly undesirable for local test modes.

Reply:
It needs to be `src/istio.io/istio`, since prow ~~will~~ should check out istio/istio for us instead of doing the clone here. 
And to your other concern, I think the clone only happens if the folder doesn't exist

Reply:
@howardjohn right, but if the repo does exist, then the e2e is done based upon the local version of Istio?  I agree with your other points.  Not sure what to do with the prefix `$(ROOT}`

Reply:
@howardjohn can you clarify your suggestion here?